### PR TITLE
added AbstractHookActivity.java file and updated remaining file accor…

### PR DESCRIPTION
**Pull request by sgupta3112**
Hi @rcvrgs , 

I have raise this PR regarding OptionSample App. 

Please review the code and let me know if any changes required. 

Shivam Gupta

### CODE REVIEW

File: OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+abstract class AbstractHookActivity extends Activity implements Observer {
+
+	private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+	protected String TAG = this.getClass().toString();
+
+	private SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+	private String text;
+	protected OptimizedOoyalaPlayerLayoutController playerLayoutController;
+	protected OoyalaPlayerLayoutController ooyalaplayerLayoutController;
+	protected String EMBEDCODE;
```
Comment (brol1dev) [inline OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: rename to `EMBED_CODE`

File: OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+
+abstract class AbstractHookActivity extends Activity implements Observer {
+
+	private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+	protected String TAG = this.getClass().toString();
+
+	private SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+	private String text;
+	protected OptimizedOoyalaPlayerLayoutController playerLayoutController;
```
Comment (brol1dev) [inline OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: use better variable names, call this one `optimizedPlayerLayoutController`

File: OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	protected OoyalaPlayerLayoutController ooyalaplayerLayoutController;
```
Comment (brol1dev) [inline OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: call this one `playerLayoutController`

**Review by brol1dev — CHANGES_REQUESTED** [review]

File: OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java (hunk 2 above)
Comment (sgupta3112) [reply OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: Done.

File: OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java (hunk 3 above)
Comment (sgupta3112) [reply OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: Done.

File: OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java (hunk 1 above)
Comment (sgupta3112) [reply OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java]: Done.

File: OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+abstract class AbstractHookActivity extends Activity implements Observer {
+
+	private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+	protected String TAG = this.getClass().toString();
+
+	private SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+	private String text;
+	protected OptimizedOoyalaPlayerLayoutController optimizedOoyalaPlayerLayoutController;
+	protected OoyalaPlayerLayoutController playerLayoutController;
```
Comment (brol1dev) [inline OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java:32]: What happens if the activity owning the `playerLayoutController` instance calls `onStop` or `onPause`? It doesn't look like the video will be paused and resumed as you do for `optimizedOoyalaPlayerLayoutController`. Test those use cases and change if there's an issue.

Comment (sgupta3112) [reply OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java:32]: Changes Done, Please review.

**Pull request merged**